src/main/java/Echo/TaskManager.java

```diff
@@ -0,0 +1,333 @@
+/*
+ * Package: Echo
+ * Module/Project Name: Echo
+ * File: TaskManager.java
+ *
+ * Description:
+ * This class manages tasks, handling operations such as adding, listing, marking, unmarking, and deleting tasks.
+ * It interacts with the Storage class to save and load tasks from a file.
+ *
+ */
+
+package Echo;
+
+import Echo.Task.Task;
+import Echo.Task.Todo;
+import Echo.Task.Deadline;
+import Echo.Task.Event;
+import Echo.Storage.Storage;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class TaskManager {
+    private List<Task> tasks;
+    private final String FILE_PATH = "." + File.separator
+            + "data" + File.separator + "echo.txt";
+    private Storage storage;
+
+    /**
+     * Constructor for the TaskManager class.
+     *
+     * @param storage The Storage object to handle file operations.
+     */
+    public TaskManager(Storage storage) {
+        this.tasks = new ArrayList<>();
+        this.storage = storage;
+        loadTasksFromFile();
+    }
+
+    /**
+     * Saves tasks to the file using the Storage class.
+     */
+    private void saveTasksToFile() {
+        storage.save(tasks);
+    }
+
+    /**
+     * Loads tasks from the file using the Storage class.
+     */
+    private void loadTasksFromFile() {
+        tasks = storage.load();
+    }
+
+    /**
+     * Executes a command based on user input.
+     *
+     * @param command The user input command.
+     */
+    public void executeCommand(String command) {
+        String[] tokens = command.split(" ", 2);
+
+        switch (tokens[0].toLowerCase()) {
```

Comment:
Perhaps tweak this such that "switch" and "case" are the same number of indents, according to the coding standard? This can be done using the IDE's settings 👍